src/types/venues.ts

```diff
@@ -25,13 +24,27 @@ export enum VenueTemplate {
   audience = "audience",
   conversationspace = "conversationspace",
   firebarrel = "firebarrel",
+  embeddable = "embeddable",
 
   /**
    * @deprecated Legacy template removed, perhaps try VenueTemplate.partymap instead?
    */
   avatargrid = "avatargrid",
 }
 
+// This type should have entries to exclude anything that has it's own specific type entry in AnyVenue below
+export type GenericVenueTemplates = Exclude<
+  VenueTemplate,
+  VenueTemplate.partymap | VenueTemplate.embeddable | VenueTemplate.jazzbar
+>;
+
+// We shouldn't include 'Venue' here, that is what 'GenericVenue' is for (which correctly narrows the types)
+export type AnyVenue =
+  | GenericVenue
+  | PartyMapVenue
+  | JazzbarVenue
+  | EmbeddableVenue;
```

Comment:
[note] All of the seemingly OOS `Venue`/`AnyVenue`/`BaseVenue`/`GenericVenue`/etc changes in this PR are related to this, which was required to narrow down the venue type passed to `Embeddable` without hacks.